### ISSUE TITLE
evaluation

### ISSUE THREAD

**Issue by zenghy96** [princeton-vl/pytorch_stacked_hourglass#13]
To generate heatmaps  on low resolution, the ground truth  was transfomed (like affine transformation) to Pts.  But if we tranform the Pts back to original resolution, there are some deviations (about 5-8 pixels) between ground truth and Pts. 
The network output  was learned from Pts, which were difference from ground truth already, did it infulence the precision final result?
Waiting for your reply! THX!

**Comment by crockwell** [comment]
I believe you are essentially correct: the model is trained on heatmaps (generated from ground truth) at lower resolution, and this probably causes very slight loss of keypoint prediction precision. The hourglass structure works at this resolution for input and output as well, and could be modified to use higher resolution at the cost of (a lot of) memory. Considering the model is pretty memory intensive, and adding more "stacks" helps results, the sacrifice in lower precision I think seemed justified given other concerns and the dataset.

 I personally did not experiment with higher resolution heatmap training, but it is possible this could make a small difference in results. Ground truths should not be biased due to this downsampling however, so I would assume the possible gains are pretty small, at least in the MPII setting. I'm curious to hear what sort of results you could get running this ablation!
